src/examples/source/SmartPeakGUI.cpp

```diff
@@ -747,80 +690,199 @@ int main(int argc, char** argv)
+      features_explorer_window_->checkbox_headers_ = session_handler_.feature_explorer_data.checkbox_headers;
+      features_explorer_window_->checkbox_columns_ = &session_handler_.feature_explorer_data.checkbox_body;
+    }
+
+    if (spectrum_explorer_window_->visible_)
+    {
+      // spectrum
+      spectrum_explorer_window_->table_data_.headers_ = session_handler_.getSpectrumExplorerHeader();
+      spectrum_explorer_window_->table_data_.body_ = session_handler_.getSpectrumExplorerBody();
+      spectrum_explorer_window_->checked_rows_ = session_handler_.spectrum_explorer_data.checked_rows;
+      spectrum_explorer_window_->checkbox_headers_ = session_handler_.spectrum_explorer_data.checkbox_headers;
+      spectrum_explorer_window_->checkbox_columns_ = &session_handler_.spectrum_explorer_data.checkbox_body;
+    }
+
+    if (calibrators_line_plot_->visible_)
+    {
+      // calibrators
+      exceeding_plot_points_ = !session_handler_.setCalibratorsScatterLinePlot(application_handler_.sequenceHandler_);
+      calibrators_line_plot_->setValues(&session_handler_.calibrators_conc_fit_data, &session_handler_.calibrators_feature_fit_data,
+        &session_handler_.calibrators_conc_raw_data, &session_handler_.calibrators_feature_raw_data, &session_handler_.calibrators_series_names,
+        session_handler_.calibrators_x_axis_title, session_handler_.calibrators_y_axis_title, session_handler_.calibrators_conc_min, session_handler_.calibrators_conc_max,
+        session_handler_.calibrators_feature_min, session_handler_.calibrators_feature_max,
+        "CalibratorsMainWindow");
+    }
+
+
+    // ======================================
+    // Windows display
+    // ======================================
+    
+    // windfow flags common to top, left and bottom windows
+    const ImGuiWindowFlags window_flags =
+      ImGuiWindowFlags_NoTitleBar |
+      ImGuiWindowFlags_NoMove |
+      ImGuiWindowFlags_NoCollapse |
+      ImGuiWindowFlags_NoFocusOnAppearing;
+
     // Left window
     if (show_left_window_) {
       ImGui::SetNextWindowPos(ImVec2(win_size_and_pos.left_window_x_pos_, win_size_and_pos.left_and_right_window_y_pos_));
       ImGui::SetNextWindowSize(ImVec2(win_size_and_pos.left_window_x_size_, win_size_and_pos.left_and_right_window_y_size_));
       ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 0);
-      const ImGuiWindowFlags left_window_flags =
-        ImGuiWindowFlags_NoTitleBar |
-        ImGuiWindowFlags_NoMove |
-        ImGuiWindowFlags_NoCollapse |
-        ImGuiWindowFlags_NoFocusOnAppearing;
-      ImGui::Begin("Left window", NULL, left_window_flags);
+      ImGui::Begin("Left window", NULL, window_flags);
       if (ImGui::BeginTabBar("Left window tab bar", ImGuiTabBarFlags_Reorderable))
       {
-        if (show_injection_explorer && ImGui::BeginTabItem("Injections", &show_injection_explorer))
+        for (auto& widget : left_windows)
```

Comment:
Wrapping all of the tabs into a `std::vector` was quite the design choice for reducing code and redundancy!